src/lib/ares_expand_name.c

```diff
@@ -52,6 +52,31 @@ static int is_reservedch(int ch)
   return 0;
 }
 
+static int ares__isprint(int ch)
+{
+  if (ch >= 0x20 && ch <= 0x7E)
+    return 1;
+  return 0;
+}
+
+/* Character set allowed by hostnames */
+static int is_hostnamech(int ch)
+{
+  /* [A-Za-z0-9-.]
+   * Don't use isalnum() as it is locale-specific
+   */
+  if (ch >= 'A' && ch <= 'Z')
+    return 1;
+  if (ch >= 'a' && ch <= 'z')
+    return 1;
+  if (ch >= '0' && ch <= '9')
+    return 1;
+  if (ch == '-' || ch == '.')
```

Comment:
`.` is not a valid character for hostnames

Reply:
function name is misleading, its validating a full domain name response which can include .